backend/RestAPI/Controllers/VideosController.cs

```diff
@@ -158,10 +159,55 @@ public async Task<ActionResult> Stream(Guid videoId, Guid userId)
                 return Unauthorized();
             }
 
-            var response = File(System.IO.File.OpenRead(video.Path), "video/mp4", enableRangeProcessing: true);
+            var response = File(System.IO.File.OpenRead(video.Path), "video/mp4",video.Title + Path.GetExtension(video.Path), enableRangeProcessing: true);
             return response;
         }
 
+        [HttpPost, Route("DownloadMultiple")]
+        public async Task<IActionResult> DownloadMultiple([FromBody] List<Guid> videoIds)
+        {
+            var userIdClaim = User.FindFirst(x => x.Type == ClaimTypes.NameIdentifier);
+            if (userIdClaim is null)
+            {
+                return NotFound();
+            }
+
+            if (!Guid.TryParse(userIdClaim.Value, out var userId))
+            {
+                return NotFound();
+            }
+
+            var user = await _usersRepository.GetUserById(userId);
+            if (user is null)
+            {
+                return NotFound();
+            }
+
+            if (videoIds.Count == 0)
```

Comment:
what about when videoIds is null?

Reply:
Not gonna pass since it is expecting an array